src/Lean/Elab/Inductive.lean

```diff
@@ -787,7 +787,7 @@ private def mkInductiveDecl (vars : Array Expr) (views : Array InductiveView) :
       for view in views do
         Term.addTermInfo' view.ref[1] (← mkConstWithLevelParams view.declName) (isBinder := true)
         for ctor in view.ctors do
-          Term.addTermInfo' ctor.ref[2] (← mkConstWithLevelParams ctor.declName) (isBinder := true)
+          Term.addTermInfo' ctor.ref[3] (← mkConstWithLevelParams ctor.declName) (isBinder := true)
```

Comment:
I can empathize. I had similar unpleasant experiences in the past. However, I am reluctant to fully move to the match-syntax style since I had even greater unpleasant experiences using it. The match-syntax has improved a lot over time, and many of these horrible experiences from the past would not happen today, but I want to move slowly to the match-syntax style.

Reply:
I'm a bit concerned about the match syntax not being zero-cost compared to using `ref[2]`. I would like to have a version of `ref[2]` that plays better with "find all references" and such (i.e. being actually tied to the syntax type involved) but isn't actually more code than this at runtime and preferably not too expensive at compile time either (I've noticed some compile time performance issues with syntax match in the past).

Reply:
@digama0 Another topic for the next dev meeting :)